Sources/StreamChat/Workers/ChannelListCleanupUpdater.swift

```diff
@@ -0,0 +1,108 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+
+/// Cleans up local channels data and refetches it from backend
+class ChannelListCleanupUpdater<ExtraData: ExtraDataTypes>: Worker {
+    // MARK: - Properties
+    
+    private let channelListUpdater: ChannelListUpdater<ExtraData>
+    
+    // MARK: - Init
+    
+    override init(
+        database: DatabaseContainer,
+        apiClient: APIClient
+    ) {
+        channelListUpdater = ChannelListUpdater(database: database, apiClient: apiClient)
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    init(
+        database: DatabaseContainer,
+        apiClient: APIClient,
+        channelListUpdater: ChannelListUpdater<ExtraData>
+    ) {
+        self.channelListUpdater = channelListUpdater
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    func cleanupChannelList() {
```

Comment:
I'm not sure we should combine resetting the `ChannelDTO` fields and the query re-fetching. What if we keep it as two separate methods?
```
func resetAllExistingChannelsData()
func refetchAllExistingChannelListQueries()
```
+ documentation of course :)